### PR TITLE
refactoring Img Segm augmentation with `albumentations`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [x] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by krshrimali** [comment]
Apologies for the delay, @Borda - this PR is on my list for today and tomorrow to review and help land if all looks good! As always, appreciate the contributions here. <3

**Comment by Borda** [comment]
> Apologies for the delay

no problem, take your time :rabbit:

File: flash/core/data/transforms.py [hunk 1]
```diff
@@ -13,11 +13,61 @@
...
+    def __init__(
+        self,
+        transform: Union[BasicTransform, Sequence[BasicTransform]],
+        mapping: dict = None,
+        image_key: str = DataKeys.INPUT,
+    ):
+        super().__init__()
+        if not isinstance(transform, (list, tuple)):
+            transform = [transform]
+        self.transform = Compose(list(transform) + [ToTensorV2()])
```
Comment (ethanwharris) [inline flash/core/data/transforms.py:46]: The added `ToTensorV2` here seems strange to me. Would prefer for the user to manually pass the tensor transform if that is something they want.

File: flash/core/data/transforms.py [hunk 2]
```diff
@@ -13,11 +13,61 @@
...
+        self._img_key = image_key
+        if not mapping:
+            mapping = self.TRANSFORM_INPUT_MAPPING
+        self._mapping_rev = mapping
+        self._mapping = {v: k for k, v in mapping.items()}
+
+    def _image_transform(self, x: Tensor) -> np.ndarray:
+        if x.ndim == 3 and x.shape[0] < 4:
```
Comment (ethanwharris) [inline flash/core/data/transforms.py:54]: Is the input always a tensor here? For e.g. image classification we use a PIL Image. Maybe we should take this opportunity to load the segmentation images and masks as PIL and NP array respectively. That way we can get rid of some of the conversion logic in here.

File: flash/core/data/transforms.py (hunk 1 above)
Comment (Borda) [reply flash/core/data/transforms.py:46]: well, I found that you pass it in each sequence so was thinking to abstract it... but if you prefer to have added by the user always, I do not mind... :rabbit:

File: flash/core/data/transforms.py (hunk 2 above)
Comment (Borda) [reply flash/core/data/transforms.py:54]: Good point, need to check some wider type expectations...

**Comment by Borda** [comment]
@krshrimali @ethanwharris may pls get your help on this collision resolution... :rabbit:

**Review by krshrimali — APPROVED** [review]
Nice! Great work, thanks @ethanwharris and @Borda. 🔥 🚀

**Pull request merged**